### PR TITLE
Typescript discussion api

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

Move discussion API to typescript.

## Does this change need to be reproduced in dotcom-rendering ?

- [X] No
- [ ] Yes (please indicate your plans for DCR Implementation)

### Tested

- [x] Locally
- [x] On CODE (**ESSENTIAL**)

#### Step-by-step `CODE`

- [x] Comments load + expand signed out
- [X] User can sign in
- [x] Comments load + expand signed in
- [x] Comments can be posted
- [x] Comments can be replied to
- [x] Comments can be recommended
- [x] Check profile comment history
- [ ] comments can be picked
- [x] comments can be reported
- [x] individual comments can be linked (however, `CODE` discussion API permalinks to `PROD`)

### CODE REVIEW

**Review by liywjl — APPROVED** [review]
LGTM 👍

File: static/src/javascripts/projects/common/modules/discussion/api.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import config from 'lib/config';
+
+type Id = number | string;
```
Comment (arelra) [inline static/src/javascripts/projects/common/modules/discussion/api.ts:3]: Can the same id be a number or string for legacy reasons?
Or are we using this type to cover two different id types in which case would it be better to have (for example) `commentId: string` and `discussionId: number` ?

File: static/src/javascripts/projects/common/modules/discussion/api.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+			},
+			credentials: 'include',
+		});
+	}
+
+	throw new Error('Discussion features have been disabled');
+};
+
+export const postComment = (
+	discussionId: number | string,
```
Comment (arelra) [inline static/src/javascripts/projects/common/modules/discussion/api.ts:43]: Could use the `Id` type here - if sticking with it?

File: static/src/javascripts/projects/common/modules/discussion/api.ts (hunk 1 above)
Comment (mxdvl) [reply static/src/javascripts/projects/common/modules/discussion/api.ts:3]: That’s a good question. I got the `string | number` union type from [the Flow archive](https://github.com/guardian/frontend/blob/6165f83791f3360fa7dc25271fc68d3da1d97f87/static/src/javascripts.flow.archive/projects/common/modules/discussion/api.js#L41). Not sure if we can confirm it should be one or the other…

**Pull request merged**